test/test_retryable_writes.py

```diff
@@ -469,6 +471,78 @@ def test_batch_splitting_retry_fails(self):
             self.assertEqual(final_txn, expected_txn)
         self.assertEqual(coll.find_one(projection={"_id": True}), {"_id": 1})
 
+    @client_context.require_multiple_mongoses
+    @client_context.require_failCommand_fail_point
+    def test_retryable_writes_in_sharded_cluster_multiple_available(self):
+        fail_command = {
+            "configureFailPoint": "failCommand",
+            "mode": {"times": 1},
+            "data": {
+                "failCommands": ["insert"],
+                "closeConnection": True,
+                "appName": "retryableWriteTest",
+            },
+        }
+
+        for mongos in client_context.mongos_seeds().split(","):
+            client = rs_or_single_client(mongos, directConnection=True)
+            set_fail_point(client, fail_command)
+
+        listener = OvertCommandListener()
+        client = rs_or_single_client(
+            client_context.mongos_seeds(),
+            appName="retryableWriteTest",
+            event_listeners=[listener],
+            retryWrites=True,
+        )
+
+        with self.assertRaises(AutoReconnect):
+            client.t.t.insert_one({"x": 1})
+
+        # Disable failpoints on each mongos
+        for mongos in client_context.mongos_seeds().split(","):
```

Comment:
Rather than opening new clients to disable the failpoint, let's save the clients that were opened above and reuse them here. Finally we should close them by registering an addCleanup(client.close) when creating them.